### DISCUSSION TITLE
Electronic Wind Controller MIDI CC2

### DISCUSSION THREAD

**Discussion by bbitwo** [FluidSynth/fluidsynth#1086] [General]
Fluidsynth is designed for keyboards with a MIDI CC11 for volume control. 
An Electronic Wind Controller (EWI) does not use CC11 but CC2 which is not evaluated by Fluidsynth. 
One ha to do some modification for evaluating the first or the other. I have some problems as I use both alternatively. 

I recommend to evaluate both simultaneously. Keyboards do not use CC2 and EWI's do not use CC11. I do not see any problem.

**Comment by mawe42** [comment]
You can use the MIDI router functionality to solve your problem. Create a config file with the following contents:
config.txt
```
router_clear

router_begin cc
router_par1 2 2 0 11
router_end

router_begin note
router_end

router_begin cc
router_end

router_begin prog
router_end

router_begin pbend
router_end

router_begin cpress
router_end

router_begin kpress
router_end
```

This clears the default routes, sets up a special cc route that turns all CC2 messages into CC11 messages. Then sets up default routes that pass all other events unchanged.
 
Feed that config file to fluidsynth with the `-f`command-line switch and you should be able to use either CC2 or CC11 for volume control.

**Comment by derselbst** [comment]
> An Electronic Wind Controller (EWI) does not use CC11 but CC2 which is not evaluated by Fluidsynth.
> I recommend to evaluate both simultaneously.

I recommend to read the documentation before opening tickets.

The [wiki](https://github.com/FluidSynth/fluidsynth/wiki/FluidFeatures#midi-control-change-implementation-chart) clearly mentions that CC2 is usable in breath mode. And it links a PDF containing comprehensive documentation about breath-, legato- and monotonic-playmodes. Page 23 will introduce breathmode, where a custom default modulator for CC2  serves as replacement for the standard note-on velocity modulator.

CC10 and CC11 always affect attenuation. If you want to change that, go with the solution Markus described.

**Reply by mawe42** [reply]
Good point... I forgot we have comprehensive EWI support.

**Reply by bbitwo** [reply]
on Apr 20, 2022  I reported my  problems of a linux dummy user. 
3 and a half years later I am still unable to use fluidsynth with my wind controller. The RTF-suggestion was wether helpful nor freindly. 
The mentioned link following up to my original mail to https://github.com/FluidSynth/fluidsynth/wiki/FluidFeatures#midi-control-change-implementation-chart was not really helpful as it only just points to 
"(002) Breath Controller 	⚠️ Usable in breathmode, see PolyMono documentation", was not really helpful. 

In the meantime I used some other soft- and hardware synths and actually tried to revist fluidsynth but still cannot find any fluidsynth setup usable from scratch for windcontrollers  for me a simple minded user.  

But I do  not give up my hope but will be back again in some years in the future. 

Kind regards

**Reply by derselbst** [reply]
You have shared very little information about your use-case and literally nothing about what you've already tried (and failed) when setting up fluidsynth for your use-case. It seems like you expect someone to look into a magic crystal ball and provide you with a tailor-made solution. As you see, this is not going to happen.

**Comment by bbitwo** [comment]
If I connect my midi usb keyboard I get get sound. If I connect my wind 
controllers with usb midi (that is my user case) I get some piano sounds 
but no usable sound with cc2. I could not activate CC2 an easy straight 
way. I was just curious how fluidsount has progressed in this 3 and a 
half years. However, value this only just as an assessment.

I did not expect a magic taylor made solution but just expected *an easy 
foolproove option, a way to switch/activate CC2* for using fluidsynth 
with my windcontroller on Linux without RTFM. I could not find such 
easily in the RTFM and gave up.

  This obviously is not possible? Looking for "fluidsynth wind 
controller" on youtube did not get a hit 2022 and does not give any 
today. May be you should think about?

Am 27.09.25 um 14:31 schrieb Tom M.:

**Reply by derselbst** [reply]
Ok, I don't have the impression that you're serious about... whatever you're trying to do. I'm not an EWI player. I know that a special breathmode was implemented by a contributor (Guitar & Piano player himself, not EWI either) to make fluidsynth EWI compatible. But I never used it myself. It can always be that the documentation we have is insufficient, inaccurate or outdated. But without a common understanding of the problem it's impossible to improve on this situation.

If you want to learn about how fluidsynth has progressed over the past 3.5 years, check our [Changelog](https://github.com/FluidSynth/fluidsynth/wiki/ChangeLog).

For the record: Our documentation is not hosted on Youtube, so this doesn't make me think.

**Comment by bbitwo** [comment]
Am 27.09.25 um 16:28 schrieb Tom M.:
> For the record: Our documentation is not hosted on Youtube, so this 
> doesn't make me think.

I have to become a little clearer for your understanding: You do not 
find any wind control player on youtube playing with fluidsynth just 
because it is not working. A wind controller has an attack and a sustain 
as long as your breath will allow, different from string instruments.

On internet you find a praise of fluidsynth by keybordplayers beside 
with notices like "My only significant problem with the program is that 
I haven't been able, so far, to get it to behave properly with my EWI 
USB wind controller. No matter how the controller is configured, the 
response to breath pressure is unpredictable and uncontrollable -- makes 
the EWI unplayable with FS.".

But, as you stated above: "this doesn't make me think". Exactly that's 
the problem I and all other wind control player have with fluidsynth and 
its developers.

Let that be my closing word. Have a good time and in a few years I check 
fluidsynth again.

Best wishes!

**Reply by derselbst** [reply]
> A wind controller has an attack and a sustain as long as your breath will allow, different from string instruments.

Fluidsynth is a SoundFont2 synthesizer. SF2 is a synth model built upon keyboards. Little surprise that this has best support. Anything beyond that must be carefully crafted into the SF2 universe.

> Exactly that's the problem I and all other wind control player have with fluidsynth and its developers.

Fluidsynth is a community effort, driven by a _single_ main-develop**er** (=me) voluntarily spending his spare time here. If our entire community only consisted of complaining and finger-pointing users like you, this project would be dead. Fortunately, this is not the case.

Your mentality however is not helpful for this project, so no need to return in a few years. Fare well.

**Reply by giuliano-simoncelli** [reply]
Hello everybody,
thank you for the information in this post!

Using my EWI controller (AKAI EWI Solo) I was experiencing a similar issue, and I found that the problem is related to how Breath Control (CC2) is interpreted.
When using Fluidsynth directly from the command line, the following command completely solves the issue:

setbreathmode 0 1 1 1

With this setting, the EWI responds correctly and the volume remains consistent between notes.

My question is:

Is there a way to apply this configuration by default (for example, by editing the user configuration file)?
I would like any external program that uses Fluidsynth (e.g., Qsynth or others) to automatically apply this setting.

Any help or suggestions would be greatly appreciated!

Regards,
Giuliano

**Comment by derselbst** [comment]
> Using my EWI controller (AKAI EWI Solo) I was experiencing a similar issue, and I found that the problem is related to how Breath Control (CC2) is interpreted.

Thanks for the information and feedback!

> Is there a way to apply this configuration by default (for example, by editing the user configuration file)?
> I would like any external program that uses Fluidsynth (e.g., Qsynth or others) to automatically apply this setting.

I would expect that if you write `setbreathmode 0 1 1 1` into any of the [config files mentioned here](https://github.com/FluidSynth/fluidsynth/wiki/FluidSynth-configuration-files#user-configuration), it should work by default. At least when using fluidsynth's commandline tool. I'm afraid Qsynth currently does not respect the default config files and doesn't seem to provide a way to load an arbitrary config file. You would have to reach out to Qsynth.

If putting `setbreathmode` into the config file does not yield the desired effect, I'd need a simple MIDI file + Soundfont to reproduce the issue.
